server/webapp/WEB-INF/rails/webpack/views/components/switch/index.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as m from "mithril";
+import {MithrilComponent} from "jsx/mithril-component";
+import {Stream} from "mithril/stream";
+
+import {bind} from "classnames/bind";
+import * as styles from "./index.scss";
+
+const uuid4      = require("uuid/v4");
+const classnames = bind(styles);
+
+export interface Attrs {
+  small?: boolean;
+  disabled?: boolean;
+  label?: string;
+  field: Stream<boolean>
+}
+
+export interface State {
+  onclick: (e: MouseEvent) => void;
+}
+
+export class Switch extends MithrilComponent<Attrs, State> {
+  oninit(vnode: m.Vnode<Attrs, State>) {
+    vnode.state.onclick = (e: MouseEvent) => {
+      const target = e.target as HTMLInputElement;
+      vnode.attrs.field(target.checked);
+    };
+  }
+
+  view(vnode: m.Vnode<Attrs, State>) {
+    const isSmall    = vnode.attrs.small;
+    const classNames = classnames({[styles.switchSmall]: isSmall}, styles.switchBtn);
+    const switchId   = `switch-${uuid4()}`;
+    const label      = vnode.attrs.label ? (
+      <label className={styles.switchLabel} data-test-id="switch-label">{vnode.attrs.label}</label>) : null;
+
+    return (
+      <div className={classNames} data-test-id="switch-wrapper">
+        {label}
+        <input id={switchId} type="checkbox"
+               {...vnode.attrs}
+               checked={vnode.attrs.field()}
+               onclick={vnode.state.onclick.bind(this)}
+               className={styles.switchInput}
+               data-test-id="switch-checkbox"/>
+        <label for={switchId} className={styles.switchPaddle} data-test-id="switch-paddle"/>
+      </div>);
+  }
```

Comment:
From what I can see, this is the same as the existing checkbox widget, with a different classname. Why redo the entire component?

Reply:
Mostly, this has differrent style then the checkbox component. We are using the switch in data sharing SPA as well(to keep ui consistent with it.)

Reply:
@ketan - I am seeing following problems with using `checkbox` component from `input_fields.tsx` -

1. The `id` attribute is not exposed in checkbox component, In switch component we need the reference of id to provide `labelFor`
2. Exposing classname will cause the inconsistance in UI as there is a way to modify the style

Not changing anything in the PR. any concerns?